operator/pkg/ciliumendpointbatch/manager.go

```diff
@@ -0,0 +1,337 @@
+		// any more new CEP's in it.
+		if len(ceb.ceb.Endpoints) >= c.maxCepsInCeb || len(ceb.ceb.Endpoints) == 0 ||
+			c.cebsToSync.isCebMarkedForDelete(ceb.ceb.GetName()) {
+			continue
+		}
+		return ceb, false
+	}
+
+	// Allocate a new ceBatch and return
+	return c.createCeb(""), true
+}
+
+// InsertCepInCache is used to Insert CEP in local cache, this may result in Creating a New
+// CEB object or Updating an existing CEB object.
+func (c *cebManagerFcfs) InsertCepInCache(cep *cilium_v2.CoreCiliumEndpoint) (string, error) {
+
+	// Check the given cep is already packed in any of the CEB.
+	// if yes, Update a ceb with the given cep object.
+	if cebName, ok := c.cacheCepMap.get(getCepNameFromCCEP(cep)); ok {
+		queueCep(cep, c.cacheCeb[cebName])
+		c.cebsToSync.updateAggregator(cebName, CebUpdate)
+		return cebName, nil
+	}
+
+	cb, isNewCeb := c.findCeb(cep)
+
+	// Cache CEP name with newly allocated CEB.
+	c.cacheCepMap.insert(getCepNameFromCCEP(cep), cb.ceb.GetName())
+
+	// Queue the CEP in CEB
+	queueCep(cep, cb)
+
+	// If it isNewCeb, update Aggregator to create a ceb
+	if isNewCeb {
+		c.cebsToSync.updateAggregator(cb.ceb.GetName(), CebCreate)
+		return cb.ceb.GetName(), nil
+	}
+
+	c.cebsToSync.updateAggregator(cb.ceb.GetName(), CebUpdate)
+	return cb.ceb.GetName(), nil
+}
+
+// RemoveCepInCache is used to remove CEP from local cache, this may result in Deleting a
+// CEB object or Updating an existing CEB object.
+func (c *cebManagerFcfs) RemoveCepFromCache(cep *cilium_v2.CoreCiliumEndpoint) error {
+
+	// Check in local cache, if a given cep is already batched in one of the ceb.
+	// and if exists, delete cep from ceb.
+	cepName := getCepNameFromCCEP(cep)
+	log.Debugf("Remove CEP from local cache :%s", cepName)
+	if cebName, ok := c.cacheCepMap.get(cepName); ok {
+		ceb, err := c.cacheCeb[cebName]
+		if err {
+			if ceb == nil || ceb.ceb == nil {
+				log.Infof("Valid cep Mapping, but no matching CEP in CEB")
+				return nil
+			}
+		}
+
+		for i, ep := range ceb.ceb.Endpoints {
+			if ep.Name == cep.Name && ep.Namespace == cep.Namespace {
+				ceb.backendMutex.Lock()
+				ceb.ceb.Endpoints =
+					append(ceb.ceb.Endpoints[:i],
+						ceb.ceb.Endpoints[i+1:]...)
+				ceb.backendMutex.Unlock()
+				break
+			}
+		}
+		log.Debugf("Removed cep:%s from ceb:%s cepCount:%d", cepName, cebName,
+			len(ceb.ceb.Endpoints))
+
+		c.cacheCepMap.deleteCep(cepName)
+
+		if len(ceb.ceb.Endpoints) == 0 {
+			log.Debugf("Remove CEB from local cache :%s", cebName)
+			c.cebsToSync.updateAggregator(cebName, CebDelete)
+			return nil
+		}
+
+		c.cebsToSync.updateAggregator(cebName, CebLazyUpdate)
+		return nil
+	}
+
+	return nil
+}
+
+// Return the total count of CEP
+func (c *cebManagerFcfs) getCepCount() int {
+	cnt := 0
+	for _, ceb := range c.cacheCeb {
+		cnt += len(ceb.ceb.Endpoints)
+	}
+	return cnt
+}
+
+// Return the total count of CEB
+func (c *cebManagerFcfs) getCebCount() int {
+	return len(c.cacheCeb)
+}
```

Comment:
Consider adding in the function doc that this function is only used in tests.

Reply:
We plan to use this function for `metrics`. Not only for `UT`.